File: backend/src/main/java/com/leavemarker/entity/Attendance.java
package com.leavemarker.entity;

import com.leavemarker.enums.AttendanceStatus;
import com.leavemarker.enums.WorkType;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalTime;

@Entity
@Table(name = "attendance",
       uniqueConstraints = {
           @UniqueConstraint(columnNames = {"employee_id", "date"})
       })
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Attendance extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "employee_id", nullable = false)
    private Employee employee;

    @Column(nullable = false)
    private LocalDate date;

    @Column
    private LocalTime punchInTime;

    @Column
    private LocalTime punchOutTime;

    @Enumerated(EnumType.STRING)
    @Column(length = 20)
    private WorkType workType;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private AttendanceStatus status;

    @Column(length = 500)
    private String remarks;

    @Column(nullable = false)
    @Builder.Default
    private Boolean correctionRequested = false;

    @Column(nullable = false)
    @Builder.Default
    private Boolean correctionApproved = false;
}

File: backend-go/internal/models/attendance.go
package models

import (
	"time"

	"github.com/leavemarker/backend-go/internal/enums"
)

// Attendance represents an attendance entity
type Attendance struct {
	BaseModel
	EmployeeID          uint                   `gorm:"not null;index;uniqueIndex:idx_emp_date" json:"employeeId"`
	Date                time.Time              `gorm:"type:date;not null;uniqueIndex:idx_emp_date" json:"date"`
	PunchInTime         *time.Time             `gorm:"type:time" json:"punchInTime"`
	PunchOutTime        *time.Time             `gorm:"type:time" json:"punchOutTime"`
	WorkType            *enums.WorkType        `gorm:"size:20" json:"workType"`
	Status              enums.AttendanceStatus `gorm:"size:20;not null" json:"status"`
	Remarks             *string                `gorm:"size:500" json:"remarks"`
	CorrectionRequested bool                   `gorm:"default:false" json:"correctionRequested"`
	CorrectionApproved  bool                   `gorm:"default:false" json:"correctionApproved"`

	// Relationships
	Employee Employee `gorm:"foreignKey:EmployeeID" json:"-"`
}

func (Attendance) TableName() string {
	return "attendances"
}
